Library/Homebrew/dev-cmd/pr-pull.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+require "utils/github"
+require "tmpdir"
+
+module Homebrew
+  module_function
+
+  def pr_pull_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `pr-pull` <pull_request>
+
+        Download and publish bottles, and apply the bottle commit from a
+        pull request with artifacts generated from GitHub Actions.
+        Requires write access to the repository.
+      EOS
+      switch "--no-publish",
+             description: "Download the bottles, apply the bottle commit, and "\
+                          "upload the bottles to Bintray, but don't publish them."
+      switch "--no-upload",
+             description: "Download the bottles and apply the bottle commit, "\
+                          "but don't upload to Bintray."
+      switch "--dry-run", "-n",
+             description: "Print what would be done rather than doing it."
+      switch "--clean",
+             description: "Do not amend the commits from pull requests."
+      switch "--workflow=",
+             description: "Specify the workflow name (default: tests.yml)."
+      switch :verbose
+      switch :debug
+      min_named 1
+    end
+  end
+
+  def signoff!(pr, path: ".", dry_run: false)
+    message = `git -C #{path} log -1 --pretty=%B`
+    close_message = "Closes ##{pr}."
+    message += "\n#{close_message}" unless message.include? close_message
+    if dry_run
+      puts "git commit --amend --signoff -m $message"
+    else
+      safe_system "git", "-C", path, "commit", "--amend", "--signoff", "--allow-empty", "-q", "-m", message
+    end
+  end
+
+  def cherry_pick_pr!(pr, path: ".", dry_run: false)
+    if dry_run
+      puts <<~EOS
+        git fetch --force origin +refs/pull/#{pr}/head
+        git merge-base HEAD FETCH_HEAD
+        git cherry-pick $merge_base..FETCH_HEAD
+      EOS
+    else
+      safe_system "git", "-C", path, "fetch", "--quiet", "--force", "origin", "+refs/pull/#{pr}/head"
+      merge_base = `git -C #{path} merge-base HEAD FETCH_HEAD`.strip
+      safe_system "git", "-C", path, "cherry-pick", "#{merge_base}..FETCH_HEAD"
+    end
+  end
+
+  def pr_pull
+    pr_pull_args.parse
+
+    if !ENV["HOMEBREW_BINTRAY_USER"] || !ENV["HOMEBREW_BINTRAY_KEY"]
+      odie "Missing HOMEBREW_BINTRAY_USER or HOMEBREW_BINTRAY_KEY variables!" unless args.dry_run? || args.no_upload?
```

Comment:
```suggestion
      odie "Missing HOMEBREW_BINTRAY_USER or HOMEBREW_BINTRAY_KEY variables!" if !args.dry_run? && !args.no_upload?
```

assuming this is correct. `unless ... ||` is a bit hard to follow.